modules/S22_php_check.sh

```diff
@@ -74,7 +82,70 @@ s22_script_check() {
     else
       COMMON_FILES_FOUND=""
     fi
-    print_output "[+] Found ""$ORANGE""parsing issues""$GREEN"" in script ""$COMMON_FILES_FOUND"":""$NC"" ""$(print_path "$LINE")" "" "$PHP_LOG"
+    print_output "[+] Found ""$ORANGE""parsing issues""$GREEN"" in script ""$COMMON_FILES_FOUND"":""$NC"" ""$(print_path "$LINE")"
     echo "$VULNS" >> "$TMP_DIR"/S22_VULNS.tmp
   fi
 }
+
+add_recommendations(){
+   local VALUE
+   local KEY
+   VALUE="$1"
+   KEY="$2"
+
+   if [[ $VALUE == *"M"* ]]; then
+      LIMIT="${VALUE//M/}"
+   fi
+
+   if [[ $KEY == *"memory_limit"* ]] && [[ $(( LIMIT)) -gt 50 ]]; then
+     return 1
+   elif [[ $KEY == *"post_max_size"* ]] && [[ $(( LIMIT)) -gt 20 ]]; then
+     return 1
+   elif [[ $KEY == *"max_execution_time"* ]] && [[ $(( LIMIT )) -gt 60 ]]; then
+     return 1
+   else
+     return 0
+   fi
+}
+
+s22_check_php_ini(){
+  mapfile -t PHP_CONF_FILES < <( find "$FIRMWARE_PATH" -xdev "${EXCL_FIND[@]}" -iname 'php.ini' -exec md5sum {} \; 2>/dev/null | sort -u -k1,1 | cut -d\  -f3 )
+  for PHP_FILE in "${PHP_CONF_FILES[@]}" ;  do
+    print_output "[*] iniscan check of ""$(print_path "$PHP_FILE")"
+    mapfile -t INISCAN_RESULT < <( "$EXT_DIR"/iniscan/vendor/bin/iniscan scan --path="$PHP_FILE" )
+    for LINE in "${INISCAN_RESULT[@]}" ; do  
+      local LIMIT_CHECK
+
+      IFS='|' read -ra LINE_ARR <<< "$LINE"
+      add_recommendations "${LINE_ARR[3]}" "${LINE_ARR[4]}"
+      LIMIT_CHECK="$?"
+      if [[ "$LIMIT_CHECK" -eq 1 ]]; then
+        print_output "$(magenta "$LINE")"
+        PHP_INI_LIMIT_EXCEEDED=$(( PHP_INI_LIMIT_EXCEEDED+1 ))
+      elif ( echo "$LINE" | grep -q "FAIL" ) && ( echo "$LINE" | grep -q "ERROR" ) ; then
+        print_output "$(red "$LINE")"
+      elif ( echo "$LINE" | grep -q "FAIL" ) && ( echo "$LINE" | grep -q "WARNING" )  ; then
+        print_output "$(orange "$LINE")"
+      elif ( echo "$LINE" | grep -q "FAIL" ) && ( echo "$LINE" | grep -q "INFO" ) ; then
+        print_output "$(blue "$LINE")"
+      elif ( echo "$LINE" | grep -q "PASS" ) ; then
+        print_output "$(green "$LINE")"
+      else
+        if ( echo "$LINE" | grep -q "failure" ) && ( echo "$LINE" | grep -q "warning" ) ; then
+          IFS=' ' read -ra LINE_ARR <<< "$LINE"
+          PHP_INI_FAILURE=${LINE_ARR[0]}
+          PHP_INI_WARNINGS=${LINE_ARR[3]}
+          print_output "$(magenta "$PHP_INI_LIMIT_EXCEEDED exceeding the limit value")""$(white ", ")""$(red "$PHP_INI_FAILURE ${LINE_ARR[1]}")""$(white " and ")""$(orange "$PHP_INI_WARNINGS ${LINE_ARR[4]}")"
+          (( S22_PHP_VULNS="$S22_PHP_VULNS"+"$PHP_INI_LIMIT_EXCEEDED"+"$PHP_INI_FAILURE"+"$PHP_INI_WARNINGS" ))
+          ((S22_PHP_SCRIPTS++))
```

Comment:
S22_PHP_SCRIPTS is for php files and not for config files. Please introduce another counter and do not mix it. Same with S22_PHP_VULNS.